### PR TITLE
[5.1] Execution of transport plugins before sending mail.

### PR DESCRIPTION
**Pull request by jaesung2061**
Reason for this change: Needed to run this package: https://github.com/fedeisas/laravel-mail-css-inliner

The current implementations of the transport classes did not do anything for the registering of plugins (empty methods registerPlugin()). It also did not execute the plugins before sending.

### CODE REVIEW

File: src/Illuminate/Mail/Transport/LogTransport.php [hunk 1]
```diff
@@ -17,6 +17,13 @@ class LogTransport implements Swift_Transport
      */
     protected $logger;
 
+	/**
+	 * Plugins for transport
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Transport/LogTransport.php]: missing fullstop.

File: src/Illuminate/Mail/Transport/LogTransport.php [hunk 2]
```diff
@@ -82,6 +91,17 @@ protected function getMimeEntityString(Swift_Mime_MimeEntity $entity)
      */
     public function registerPlugin(Swift_Events_EventListener $plugin)
     {
-        //
+		array_push($this->plugins, $plugin);
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Transport/LogTransport.php]: you MUST indent with spaces everywhere

File: src/Illuminate/Mail/Transport/LogTransport.php [hunk 3]
```diff
@@ -82,6 +91,17 @@ protected function getMimeEntityString(Swift_Mime_MimeEntity $entity)
...
     }
+
+	/**
+	 * @param $message
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Transport/LogTransport.php]: loads of missing info from this phpdoc

File: src/Illuminate/Mail/Transport/MailgunTransport.php [hunk 4]
```diff
@@ -38,6 +38,13 @@ class MailgunTransport implements Swift_Transport
      */
     protected $url;
 
+	/**
+	 * Plugins for transport
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Transport/MailgunTransport.php]: cs

File: src/Illuminate/Mail/Transport/MailgunTransport.php [hunk 5]
```diff
@@ -104,9 +113,20 @@ public function send(Swift_Mime_Message $message, &$failedRecipients = null)
...
+		array_push($this->plugins, $plugin);
     }
 
+	/**
+	 * @param $message
+	 */
+	protected function beforeSendPerformed(Swift_Mime_Message $message)
+	{
+		foreach ($this->plugins as $plugin) {
+			$evt = new \Swift_Events_SendEvent($this, $message);
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Transport/MailgunTransport.php]: you MUST import this

**Comment by jaesung2061** [comment]
First pull request ever, sorry for the messy code.

File: src/Illuminate/Mail/Transport/LogTransport.php [hunk 6]
```diff
@@ -82,6 +92,19 @@ protected function getMimeEntityString(Swift_Mime_MimeEntity $entity)
...
     {
-        //
+        array_push($this->plugins, $plugin);
+    }
+
+    /**
+     * Iterate through registered plugins and execute plugins' methods.
+     *
+     * @param Swift_Mime_Message $message
+     */
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Transport/LogTransport.php]: You need a `@return void` here.

File: src/Illuminate/Mail/Transport/MailgunTransport.php [hunk 7]
```diff
@@ -104,7 +114,20 @@ public function send(Swift_Mime_Message $message, &$failedRecipients = null)
...
     {
-        //
+        array_push($this->plugins, $plugin);
+    }
+
+    /**
+     * Iterate through registered plugins and execute plugins' methods.
+     *
+     * @param Swift_Mime_Message $message
+     */
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Transport/MailgunTransport.php]: You need a `@return void` here.

File: src/Illuminate/Mail/Transport/MandrillTransport.php [hunk 8]
```diff
@@ -136,4 +159,5 @@ public function setKey($key)
     {
         return $this->key = $key;
     }
+
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Transport/MandrillTransport.php]: no

File: src/Illuminate/Mail/Transport/SesTransport.php [hunk 9]
```diff
@@ -70,7 +80,20 @@ public function send(Swift_Mime_Message $message, &$failedRecipients = null)
...
     {
-        //
+        array_push($this->plugins, $plugin);
+    }
+
+    /**
+     * Iterate through registered plugins and execute plugins' methods.
+     *
+     * @param Swift_Mime_Message $message
+     */
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Transport/SesTransport.php]: and everywhere else too

**Comment by taylorotwell** [comment]
I DRY'd up the code some. Can you give it a test?

**Comment by jaesung2061** [comment]
It's working great!

**Pull request merged**